src/main/java/codeu/controller/MyProfileServlet.java

```diff
@@ -18,23 +19,55 @@
 * Servlet class responsible for profile feed.
 */
 public class MyProfileServlet extends HttpServlet {
-	/**
-  	* Store class that gives access to Users.
-  	*/
- 	private UserStore userStore;
 
- @Override
- public void doGet(HttpServletRequest request, HttpServletResponse response)
-     throws IOException, ServletException {
+  /** Store class that gives access to Users. */
+  private UserStore userStore;
 
-   request.getRequestDispatcher("/WEB-INF/view/myprofile.jsp").forward(request, response);
- }
   /**
-  * Sets the UserStore used by this servlet. This function provides a common setup method
-  * for use by the test framework or the servlet's init() function.
-  */
- void setUserStore(UserStore userStore) {
-   this.userStore = userStore;
- }
+   * Set up state for handling login-related requests. This method is only called when running in a
+   * server, not when running in a test.
+   */
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setUserStore(UserStore.getInstance());
+  }
 
-}
\ No newline at end of file
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+	
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+   throws IOException, ServletException {
+       request.getRequestDispatcher("/WEB-INF/view/myprofile.jsp").forward(request, response);
+  }
+	
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+  throws IOException, ServletException {
+	 String username = (String) request.getSession().getAttribute("user");
+	 User user = userStore.getUser(username);
+	 String photo_url = request.getParameter("photo_url");
+
+	 if (user != null) {
+	   user.setPhotoURL(photo_url);
+	     try {
+		   userStore.updateUserPhoto(user, photo_url);
+		 }
+		   catch (Exception e) {
+		   System.out.println(e);	
+		 }	
+       request.getSession().setAttribute("photo_url", user.getPhotoURL());
+       response.sendRedirect("/myprofile");
+	 }
+	 else {  //user object not found - if not logged in, can't change the picture
+	   request.setAttribute("error", "Error, please login.");
```

Comment:
Since they have to be logged in to set profile pic, I don't know if this else case is needed but we do want to redirect them to myprofile right?

Reply:
I added it because if the user types in /myprofile in the url (although they can't see it in navbar, you can access the page by typing it in so), user will see the profile page. It just wouldn't have any picture or info. So if somehow user gets access to profile page and tries to update something it will crash. So to avoid the website of crashing, user will see this error message instead.